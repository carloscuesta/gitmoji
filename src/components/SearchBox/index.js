// @flow
import React, { useContext } from 'react'
import { store } from '../Store/store.js'

type Props = { value?: string }

const SearchBox = (props: Props) => {
  const globalState = useContext(store)
  const { dispatch } = globalState

  const onChange = (data) => {
    dispatch({ type: 'new value', payload: data.target.value })
  }
  return (
    <input
      maxLength="70"
      type="text"
      onChange={onChange}
      placeholder="Search your emoji"
    />
  )
}

export default SearchBox
