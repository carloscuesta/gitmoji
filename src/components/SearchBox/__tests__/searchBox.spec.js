import renderer from 'react-test-renderer'

import SearchBox from '../index'

describe('SearchBox', () => {
  it('should render the component', () => {
    const wrapper = renderer.create(<SearchBox />)
    expect(wrapper).toMatchSnapshot()
  })
})
