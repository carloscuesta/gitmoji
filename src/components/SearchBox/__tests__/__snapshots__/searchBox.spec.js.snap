// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render the component 1`] = `
<input
  maxLength="70"
  onChange={[Function]}
  placeholder="Search your emoji"
  type="text"
/>
`;
